File: src/Forjix.Application/Features/Sales/SalesModels.cs
namespace Forjix.Application.Features.Sales;

public sealed record CreateSaleItemRequest(Guid ProductId, decimal Quantity);
public sealed record CreateSaleRequest(string PaymentMethod, decimal Discount, Guid? CustomerId, IReadOnlyList<CreateSaleItemRequest> Items);
public sealed record CancelSaleRequest(string Reason, string RowVersion);
public sealed record SaleItemView(Guid Id, Guid ProductId, string ProductName, string Sku, decimal Quantity, decimal UnitPrice, decimal UnitCost, decimal Discount, decimal Total);
public sealed record SaleView(Guid Id, string Number, string Status, decimal Subtotal, decimal Discount, decimal Total, string PaymentMethod, Guid UserId, string UserName, Guid? CustomerId, DateTimeOffset CreatedAt, DateTimeOffset? CancelledAt, string? CancellationReason, string RowVersion, IReadOnlyList<SaleItemView> Items);
public sealed record SaleListItem(Guid Id, string Number, string Status, decimal Total, string PaymentMethod, string UserName, Guid? CustomerId, int ItemCount, DateTimeOffset CreatedAt, string RowVersion);
public sealed record PagedSales(IReadOnlyList<SaleListItem> Items, int Page, int PageSize, int Total);

File: web/forjix-web/src/app/core/api/sales.models.ts
export type PaymentMethod = 'Cash' | 'Pix' | 'CreditCard' | 'DebitCard';
export interface CreateSaleRequest { paymentMethod: PaymentMethod; discount: number; customerId?: string | null; items: { productId: string; quantity: number }[]; }
export interface SaleItem { id: string; productId: string; productName: string; sku: string; quantity: number; unitPrice: number; unitCost: number; discount: number; total: number; }
export interface Sale { id: string; number: string; status: 'Completed' | 'Cancelled'; subtotal: number; discount: number; total: number; paymentMethod: PaymentMethod; userId: string; userName: string; customerId?: string; createdAt: string; cancelledAt?: string; cancellationReason?: string; rowVersion: string; items: SaleItem[]; }
export interface SaleListItem { id: string; number: string; status: 'Completed' | 'Cancelled'; total: number; paymentMethod: PaymentMethod; userName: string; customerId?: string; itemCount: number; createdAt: string; rowVersion: string; }
export interface PagedSales { items: SaleListItem[]; page: number; pageSize: number; total: number; }
